### DISCUSSION TITLE
Learning Resources for Helix

### DISCUSSION THREAD

**Discussion by haxfn** [helix-editor/helix#10925] [General]
#### I am really amazed by Helix and its modern outlook. I have been using Vim and Neovim for last 2 years. I am really amazed by these things:

- Everything just works out of the box
- Improved and rethought vim-like motions and actions
- emacs like keybinds at some places
- upcoming merging of lisp-based plugins

Helix is great and will be accepted and integrated at more and more places(browsers, shells, ides), would love to contribute to this end.

#### Given this, I want to pickup more things about Helix.  

The tutor doesn't do justice, I have found the following, but I request links to further my learning of the Helix keybinds.

1. https://docs.helix-editor.com/keymap.html
2. https://tomgroenwoldt.github.io/helix-shortcut-quiz/
3. https://github.com/stevenhoy/helix-cheat-sheet
4. https://quantonganh.com/2023/08/19/turn-helix-into-ide

Some things I really found useful:
```
1. command_pallete :: <space>? (space+?) to search commands/keybinds, use keywords (don't use phrases)
2. symbol_picker   :: <space>s to goto code blocks, to move to symbols
3. file_picker     :: <space>f to open the file picker
```

Please suggest more learning resources.

**Comment by stevenhoy** [comment]
I made this [cheat sheet](https://github.com/stevenhoy/helix-cheat-sheet) - perhaps it will help? 
It's for v23 but I plan to update it to v24.
S.

**Reply by haxfn** [reply]
useful!

**Reply by richardhttps** [reply]
Thanks!

**Reply by igorbabko** [reply]
Thank you!

**Comment by David-Else** [comment]
https://helix-editor-tutorials.com/tutorials/

**Comment by tomgroenwoldt** [comment]
I reworked https://keygli.de, it's based on the helix editor. Feel free to check it out :)

**Comment by xpe** [comment]
@tomgroenwoldt KeyGlide looks really nice. I was just wanting this!

Do you have a preferred place for feature requests and/or donations? In the meanwhile, I would like to mention two things:

1. Personally, I'd prefer this gamification experience in my local Helix editor, offline, without Internet access. Like a beefed up `hx --tutorial` maybe. I recognize this might not be possible since Helix doesn't have plugins... yet. It might require forking for now?

2. If I do use [keyglide](https://keygli.de) (in a web browser), it doesn't quite fit for me., since I use Colemak and have remapped a lot of keys.

**Reply by tomgroenwoldt** [reply]
Thanks for your feedback :) I'll open a public issue board soon. Here are my thoughts:

1.  I understand and I briefly worked on something that aimed for this to be entirely local. But (at least for me) it turned out to be unfeasible as the application was not in charge of the editor. It's a lot harder to control a user's local editor than something running on your server.
2. I plan to support custom configs where users are allowed to upload their helix configurations to their profile. This would also imply making this configuration public to other users as otherwise it would be hard to understand the keystrokes.

Update: https://github.com/tomgroenwoldt/keyglide-feedback

**Reply by tomgroenwoldt** [reply]
Hm, currently thinking about adding custom configuration support. Though, this might break the whole purpose of having a universal solution for everyone. Or do you think one could remap a solution you typed via your Colemak remap again - back to the "original"?
